smithy-openapi/src/main/java/software/amazon/smithy/openapi/fromsmithy/protocols/AbstractRestProtocol.java

```diff
@@ -209,16 +210,39 @@ private boolean needsInlineTimestampSchema(Context<? extends Trait> context, Mem
                 param.style("form").explode(true);
             }
 
-            // Create the appropriate schema based on the shape type.
-            Schema refSchema = context.inlineOrReferenceSchema(member);
-            QuerySchemaVisitor<T> visitor = new QuerySchemaVisitor<>(context, refSchema, member);
-            param.schema(target.accept(visitor));
+            param.schema(createQuerySchema(context, member, target));
+            result.add(param.build());
+        }
+
+        for (HttpBinding binding : httpBindingIndex.getRequestBindings(operation, HttpBinding.Location.QUERY_PARAMS)) {
```

Comment:
nit: this code is super similar to the loop above. The difference being that when using httpQueryParams it's always a form.